donjo-app/models/First_artikel_m.php

```diff
@@ -474,10 +474,12 @@ public function list_sosmed()
 
 	public function hit($url)
 	{
+		$this->load->library('user_agent');
+
 		$this->db->where('id', $url)->or_where('slug', $url);
 		$id = $this->db->select('id')->get('artikel')->row()->id;
 		//membatasi hit hanya satu kali dalam setiap session
-		if (in_array($id, $_SESSION['artikel'])) return;
+		if (in_array($id, $_SESSION['artikel']) OR $this->agent->is_robot()) return;
```

Comment:
Perhatikan komentar di panduan CI, bahwa daftar robot mereka belum tentu lengkap. Dan kalau perlu bisa dilengkapi di `application/config/user_agents.php`. Mungkin nanti (tidak harus sekarang) perlu ditambah script untuk mendeteksi robot lain. Mungkin info di https://stackoverflow.com/questions/17515381/exclude-bots-and-spiders-from-a-view-counter-in-php yg dibicarakan di group Telegram (dan acuan lain) bisa memberi info mengenai pola robot lain yg belum terdeteksi oleh CI.

Reply:
Mungkin kondisinya bisa dibalik pak @eddieridwan ??? Cek device/browser selain itu dibaca robot. 

Sy lihat listnya terlalu banyak kalau mau dimasukkan:

https://github.com/monperrus/crawler-user-agents/blob/master/crawler-user-agents.json

https://github.com/JayBizzle/Crawler-Detect/blob/c47ab0b8a63897265f692a82d638bd14ca28a807/src/Fixtures/Crawlers.php

Reply:
Periksa apa yg sdh disediakan di CI. Tambahkan yg belum saja. Batasi awalnya untuk pola umum saja dulu -- nanti bisa ditambahkan robot spesifik berdasarkan pengalaman. Tapi juga penyempurnaan ini tdk perlu sekarang. Langkah berikutnya mungkin untuk fitur mencatat pengunjung yg berulang-ulang -- sebagai langkah untuk deteksi robot. Sekali lagi, tidak perlu untuk PR ini.

Reply:
Atau mungkin lebih baik baca langsung dari https://github.com/monperrus/crawler-user-agents/blob/master/crawler-user-agents.json. Jadi memanfaatkan usaha banyak orang yg mendaftarkan setiap crawler yg ditemukan. Simpan di database di tabel crawler. Dan hanya update secara berkala kalau daftar itu berubah. Proses untuk update tabel crawler itu bisa dimasukkan di migrasi.

Reply:
Di temuan bung @afa28 , di https://github.com/monperrus/crawler-user-agents/, ada beberapa library PHP alternatif yg sepertinya siap pakai untuk men-deteksi crawler. Mungkin bisa jajaki untuk melihat apakah ada yg memadai /sdh mencukupi untuk OpenSID.